### PR TITLE
DataSource: Add an option to get hidden queries in plugin.json

### LINKED ISSUE #17123: Do not filter hidden queries in PanelQueryRunner
We cannot filter out hidden queries outside the data source as some data source support referring to hidden queries. 

Screen Shot 2019-05-16 at 20 52 52

@ryantxu sorry, it was a good idea but not sure we can do this.. We could keep it as is and have an opt-in option for data sources to say they want to handle hidden queries but that would break some external data sources that also support this (Meta Queries and a few others I think).

### PR DESCRIPTION
**Pull request by ryantxu**
Fixes #17123 -- Totally untested, but maybe a good place to start

This keeps the default behavior as cloned and filtered queries, but gives you the option to get all of them

### CODE REVIEW

**Review by torkelo — COMMENTED** [review]
Waiting to merge this until we add processHiddenQueries to Graphite & Cloudwatch data sources (who both need hidden queries as other queries can refer to them)

File: packages/grafana-ui/src/types/datasource.ts [hunk 1]
```diff
@@ -84,6 +84,11 @@ export interface DataSourcePluginMeta extends PluginMeta {
   category?: string;
   queryOptions?: PluginMetaQueryOptions;
   sort?: number;
+
+  /**
+   * By default, hidden queries are not passed to the datasource
+   */
+  processHiddenQueries?: boolean;
```
Comment (torkelo) [inline packages/grafana-ui/src/types/datasource.ts]: always find verb / action named properties that are bool flags wrong but was hard pressed come up with an alterantive here. 

* needsHiddenQueries
* handlesHiddenQueries
* ?

Comment (ryantxu) [reply packages/grafana-ui/src/types/datasource.ts]: maybe supportsHiddenQueries?

**Comment by ryantxu** [comment]
I updated this with supportsHiddenQueries on Graphite & Cloudwatch --- still totally untested, I have never used Graphite or Cloudwatch :)

**Comment by torkelo** [comment]
Decided on just "hiddenQueries" flag as all the other feature flags where single word names "metrics", "annotations", "alerting" etc.

**Review by torkelo — APPROVED** [review]

**Pull request merged**